code/modules/antagonists/traitor/contractor/items/contractor_pinpointer.dm

```diff
@@ -0,0 +1,56 @@
+/obj/item/pinpointer/crew/contractor
+	name = "contractor pinpointer"
+	desc = "A handheld tracking device that points to crew without needing suit sensors at the cost of accuracy."
+	icon_state = "pinoff_contractor"
+	icon_off = "pinoff_contractor"
+	icon_null = "pinonnull_contractor"
+	icon_direct = "pinondirect_contractor"
+	icon_close = "pinonclose_contractor"
+	icon_medium = "pinonmedium_contractor"
+	icon_far = "pinonfar_contractor"
+	/// The minimum range for the pinpointer to function properly.
+	var/min_range = 20
+	/// The first person to have used the item. If this is set already, no one else can use it.
+	var/mob/owner = null
+
+/obj/item/pinpointer/crew/contractor/point_at(atom/target, spawnself = TRUE)
+	if(!active)
+		return
+	if(!target)
+		icon_state = icon_null
+		return
+
+	var/turf/T = get_turf(target)
+	var/turf/L = get_turf(src)
+
+	if(!(T && L) || (T.z != L.z))
+		icon_state = icon_null
+	else
+		dir = get_dir(L, T)
+		var/dist = get_dist(L, T)
+		if(ISINRANGE(dist, -1, min_range))
+			icon_state = icon_direct
+		else if(ISINRANGE(dist, min_range + 1, min_range + 8))
+			icon_state = icon_close
+		else if(ISINRANGE(dist, min_range + 9, min_range + 16))
+			icon_state = icon_medium
+		else if(ISINRANGE(dist, min_range + 16, INFINITY))
+			icon_state = icon_far
+	if(spawnself)
+		spawn(0.5 SECONDS)
+			.()
```

Comment:
.()?
Oh it's to infinite loop itself and keep calling itself.... well okay it's how the other pinpointer works I guess.
Can you however refactor this one to have it not be copy paste from the other and instead extract the differences into procs?